lib/business_time/time_extensions.rb

```diff
@@ -194,5 +197,34 @@ def consecutive_days
       end
       (days << self).sort
     end
+
+    def to_array_of_dates(passed_value)
+      passed_value ||= []
+      array_of_dates = []
+
+      case passed_value
+        when Date
+          array_of_dates = [passed_value]
+        when Time
+          array_of_dates = [passed_value.to_date]
+        when String
+          array_of_dates = [Date.parse(passed_value)]
+        when Array
+          passed_value.each do |holiday|
+            case holiday
+              when String
+                array_of_dates << Date.parse(holiday)
+              when Time
+                array_of_dates << holiday.to_date
+              when Date
+                array_of_dates << holiday
+              else
+                raise ArgumentError.new("Holidays specified must be a Date, Time or String.")
+            end
+          end
+      end
+
+      array_of_dates
+    end
```

Comment:
[SUGGESTION] I think this entire `#to_array_of_dates` method can be distilled down to something much simpler. For example:

```ruby
def to_array_of_dates(values)
  Array.wrap(values).map(&:to_date)
end
```

`Array.wrap` wraps it's arg in an array unless it's already an array.

`String`, `Time`, and `Date` all respond to `#to_date` (thanks to ActiveSupport).   If someone passes something that doesn't quack with the `#to_date` method, let the `NoMethodError` bubble up and ignore the `ArgumentError` check.

Reply:
@rmm5t Very cool, just committed the changes 👍